src/index.ts

```diff
@@ -0,0 +1,518 @@
+import LanguageTag from './utils/LanguageTag';
+import {
+  PluralCategory,
+  arabic,
+  eastSlavic,
+  oneOther,
+  oneTwoOther,
+  oneUpToTwoOther,
+  oneWithZeroOther,
+  other,
+  polish,
+  westSlavic,
+} from './utils/pluralization';
+
+// Define some variables for validating keys and key context.
+const CONTEXT_CHARACTERS = '[A-Za-z0-9-_]+';
+const DELIMITER = '.';
+
+const VALID_KEY_CONTEXT = new RegExp(`^${CONTEXT_CHARACTERS}$`);
+const VALID_KEY = new RegExp(`^${CONTEXT_CHARACTERS}(${DELIMITER}${CONTEXT_CHARACTERS})*$`);
+
+// Used to find variable names inside the output string.
+const INTERPOLATION_REGEX = /%\{([^}]*)\}/g;
+
+// Default locale is english.
+const DEFAULT_LOCALE = 'en';
+
+// Export PluralCategory so it is easy for others to create rules.
+export { PluralCategory };
+
+interface Aliases {
+  [propName: string]: string;
+};
+
+export interface Translations {
+  [propName: string]: Translations | string;
+};
+
+export interface Rule {
+  (n: number): PluralCategory;
+};
+
+interface Rules {
+  [propName: string]: Rule;
+}
+
+interface SetLocaleOptions {
+  ignorePreferenceChain?: boolean;
+}
+
+export interface TDataObject {
+  count?: number;
+  [propName: string]: string | number | undefined;
+};
+
+export interface TOptions {
+  fallback?: string;
+  locale?: string;
+}
+
+export interface Unsubscribe {
+  (): void;
+}
+
+type TranslationLookup = Translations | string | undefined;
```

Comment:
I don't think this type is necessary and the name doesn't suggest a concrete thing to me. For the usage in `t` it can be inferred, and for the return value of the reducer, you can use index syntax
```typescript
  private translationReducer(translations: Translations, context: string): Translations[string] { ... }
```

Somewhat objective though. If you disagree, consider instead using the index syntax for the alias.

Reply:
This type actually felt the most awkward to me. I was unaware that you could specify the return type like that. That's a pretty nice feature.